src/haddock/modules/scoring/__init__.py

```diff
@@ -127,47 +137,121 @@ def per_interface_output(
+        ]
+        return combinations
+
+    @staticmethod
+    def compute_interfaces_score(
+            interface_energies: dict[str, dict[str, float]],
+            interface_sets_combinations: list[str],
+            ) -> Optional[float]:
+        """Compute the sum of selected interface haddock score.
+
+        Parameters
+        ----------
         interfaces_scores : dict[str, dict[str, float]]
-            Dictionary holding per interfaces scores.
+            Scores of the various interfaces present in the pdb.
+        interface_sets_combinations : list[str]
+            List of interface combinations to consider
         """
-        header = None
-        interfaces_scores: dict[str, dict[str, float]] = {}
-        with open(pdb.file_name, 'r') as filin:
-            for _ in filin:
-                if _.startswith('REMARK Interface'):
-                    s_ = _.strip().split()[2:]
-                    # Extract header
-                    if not header:
-                        header = s_
-                    # Extract data
-                    else:
-                        chain1 = s_[header.index('Chain1')]
-                        chain2 = s_[header.index('Chain2')]
-                        haddockscore = float(s_[header.index('HADDOCKscore')])
-                        evdw = float(s_[header.index('Evdw')])
-                        eelec = float(s_[header.index('Eelec')])
-                        edesol = float(s_[header.index('Edesol')])
-                        bsa = float(s_[header.index('BSA')])
-                        # Combine chains together
-                        chains_key = f"{chain1}_{chain2}"
-                        # Hold data
-                        interfaces_scores[chains_key] = {
-                            'HADDOCKscore': haddockscore,
-                            'Evdw': evdw,
-                            'Eelec': eelec,
-                            'Edesol': edesol,
-                            'BSA': bsa,
-                            }
-        return interfaces_scores
+        # Minimum set of interfaces must be >= 1
+        if len(interface_sets_combinations) < 1:
+            return None
+
+        # Get all desired interfaces scores
+        selected_interfaces_scores: list[float] = []
+        for interface in interface_sets_combinations:
+            if interface in interface_energies.keys():
+                interface_score = interface_energies[interface]["HADDOCKscore"]
+                selected_interfaces_scores.append(interface_score)
+            else:
+                # Check if reverse chain order present
+                split_inter = interface.split("_")
+                reverse_interface = f"{split_inter[1]}_{split_inter[0]}"
+                if reverse_interface in interface_energies.keys():
+                    interface_score = interface_energies[reverse_interface]["HADDOCKscore"]
+                selected_interfaces_scores.append(interface_score)
```

Comment:
Variable `interface_score` can be unbound here as it is defined inside a nested if/else condition, so there can be the case in which the condition is not triggered and this will raise a `UnboundLocalError`.

`interface_score`  here must be have a zero-value defined outside the if/else conditional to avoid the unbound error - but would be better to refactor this block.